caddy_test.go

```diff
@@ -56,3 +59,38 @@ func TestIsLoopback(t *testing.T) {
 		}
 	}
 }
+
+func TestListenerAddrEqual(t *testing.T) {
+	ln1, err := newLocalListener("[::]:2016")
```

Comment:
I use ports 2017 and 2018 in development sometimes (silly, I know) -- would probably be better to use port 0 and then use ln.Address() to get the address of the listener. Using port 0 will let the system assign a port that's available.